### PR TITLE
Fixes --remote flag issues

### LINKED ISSUE #6598: Non-obvious extra argument to new podman --remote
## current (V1)

systemctl --user start io.podman.socket
export PODMAN_VARLINK_ADDRESS=unix:/run/user/1000/podman/io.podman

```console
 podman version
Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.13.8
OS/Arch:            linux/amd64
 podman-remote version
Client:
Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.13.8
OS/Arch:            linux/amd64

Server:
Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.13.8
OS/Arch:            linux/amd64
```

## development (V2)

systemctl --user start podman.socket
\# hacked the podman.service unit to run /usr/local/bin/podman2 instead of /usr/bin/podman

```console
 podman2 version
Version:      2.0.0-dev
API Version:  1
Go Version:   go1.13.8
Git Commit:   3f026eb6a682a68e69f9376b72157a8f084e575c
Built:        Sat Jun 13 17:53:06 2020
OS/Arch:      linux/amd64
 podman2 --remote version
Error: unrecognized command `podman2 version`
Try 'podman2 --help' for more information.
 podman2 --remote podman version
Client:
Version:      2.0.0-dev
API Version:  1
Go Version:   go1.13.8
Git Commit:   3f026eb6a682a68e69f9376b72157a8f084e575c
Built:        Sat Jun 13 17:53:06 2020
OS/Arch:      linux/amd64

Server:
Version:      2.0.0-dev
API Version:  0
Go Version:   go1.13.8
Git Commit:   3f026eb6a682a68e69f9376b72157a8f084e575c
Built:        
OS/Arch:      linux/amd64
```

So it seems that I have to give an extra argument, after the `-r` ?

Seems like the actual value doesn't really matter (foo, bar, etc)

### LINKED ISSUE #6704: location of the `--remote` flag breaks commands.
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The location of the `--remote` flag changes the behavior of the command line parsing.

**Steps to reproduce the issue:**

```bash
➜  ~ podman system service -t 0 & 
[1] 65413
➜  ~ podman ps --remote
CONTAINER ID  IMAGE   COMMAND  CREATED  STATUS  PORTS   NAMES
➜  ~ podman --remote ps
Error: unrecognized command `podman ps`
Try 'podman --help' for more information.
➜  ~ kill %
[1]  + 65413 terminated  podman system service -t 0    
```

**Describe the results you received:**

`podman --remote ps` doesn't work

**Describe the results you expected:**

 `podman ps --remote` and `podman --remote ps` should be equivalent.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**
```bash
➜  ~ podman version
Version:      2.0.0
API Version:  1
Go Version:   go1.14.4
Git Commit:   eec482cae3289ecaad45c602629657da7062ce9c
Built:        Sat Jun 20 14:30:13 2020
OS/Arch:      linux/amd64
```
**Output of `podman info --debug`:**

```
➜  ~ podman info --debug
host:
  arch: amd64
  buildahVersion: 1.15.0
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/bin/conmon
    version: 'conmon version 2.0.18, commit: 7b3e303be8f1aea7e0d4a784c8e64a75c14756a4'
  cpus: 4
  distribution:
    distribution: arch
    version: unknown
  eventLogger: file
  hostname: carbon
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.7.4-arch1-1
  linkmode: dynamic
  memFree: 2045399040
  memTotal: 8218943488
  ociRuntime:
    name: runc
    package: Unknown
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc10
      commit: dc9208a3303feef5b3839f4323d9beb36df0a9dd
      spec: 1.0.1-dev
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.1.1
      commit: bbf27c5acd4356edb97fa639b4e15e0cd56a39d5
      libslirp: 4.3.0
      SLIRP_CONFIG_VERSION_MAX: 3
  swapFree: 0
  swapTotal: 0
  uptime: 1h 30m 43.76s (Approximately 0.04 days)
registries:
  search:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  configFile: /home/tino/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.9.2
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.2
        using FUSE kernel interface version 7.31
  graphRoot: /home/tino/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 366
  runRoot: /run/user/1000
  volumePath: /home/tino/.local/share/containers/storage/volumes
version:
  APIVersion: 1
  Built: 1592656213
  BuiltTime: Sat Jun 20 14:30:13 2020
  GitCommit: eec482cae3289ecaad45c602629657da7062ce9c
  GoVersion: go1.14.4
  OsArch: linux/amd64
  Version: 2.0.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**
```
➜  ~ pacman -Q podman
podman 2.0.0-1
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

tested on my local laptop running archlinux.

### PR DESCRIPTION
**Pull request by jwhonce**
* --remote, --url and --identity are now anchored to podman command.
      Subcommands should no longer have issues
    * TraverseChildren now set to V1 expectations
    * Latest flag now has helper function. Now has consistent usage.
    * IsRemote() uses cobra parser to determin if --remote is given
    * Moved validation functions from parser pkg to validate pkg

Fixes #6598
Fixes #6704

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: cmd/podman/diff.go [hunk 1]
```diff
@@ -17,12 +17,11 @@ var (
...
 	diffCmd         = &cobra.Command{
-		Use:              "diff [flags] {CONTAINER_ID | IMAGE_ID}",
-		Args:             validate.IDOrLatestArgs,
-		Short:            "Display the changes of object's file system",
-		Long:             diffDescription,
-		TraverseChildren: true,
-		RunE:             diff,
+		Use:   "diff [flags] {CONTAINER_ID | IMAGE_ID}",
+		Args:  validate.IDOrLatestArgs,
+		Short: "Display the changes of object's file system",
```
Comment (TomSweeneyRedHat) [inline cmd/podman/diff.go:22]:
```suggestion
		Short: "Display the changes to the object's file system",
```

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@jwhonce Tests are not happy.

**Comment by mheon** [comment]
LGTM

**Pull request merged**